InnerEye/ML/Histopathology/utils/metrics_utils.py

```diff
@@ -97,3 +101,67 @@ def plot_slide_noxy(slide: str, score: float, paths: List, attn: List, case: str
     for i in range(len(axs.ravel())):
         axs.ravel()[i].set_axis_off()
     return fig
+
+
+def plot_slide(slide_image: np.ndarray, scale: float) -> plt.figure:
+    """Plots a slide thumbnail from a given slide image and scale.
+    :param slide_image: Numpy array of the slide image (shape: [3, H, W]).
+    :return: matplotlib figure of the slide thumbnail.
+    """
+    fig, ax = plt.subplots()
+    slide_image = slide_image.transpose(1, 2, 0)
+    ax.imshow(slide_image)
+    ax.set_axis_off()
+    original_size = fig.get_size_inches()
+    fig.set_size_inches((original_size[0]*scale, original_size[1]*scale))
+    return fig
+
+
+def plot_heatmap_overlay(slide: str, 
+                       slide_image: np.ndarray,
+                       results: Dict[str, List[Any]],
+                       location_bbox: List[int],
+                       tile_size: int = 224,
+                       level: int = 1) -> plt.figure:
+    """Plots heatmap of selected tiles (e.g. tiles in a bag) overlay on the corresponding slide.
+    :param slide: slide identifier.
+    :param slide_image: Numpy array of the slide image (shape: [3, H, W]).
+    :param results: Dict containing ResultsKey keys (e.g. slide id) and values as lists of output slides.
```

Comment:
Another design comment: if slide id is in this dictionary already there is no need to pass the slide identifier, you could have passed to this function only the part of interest of the dictionary (results[:][slide_id]) and this function would have been much more generic and decoupled from possible refactoring of the results dictionary.  In the interest of getting this merged is OK not to implement this change but something to keep in mind for the next time!

Reply:
Ok, thanks. I will keep in mind next time.